cmd/zed/agents/fmd_api.h

```diff
@@ -147,8 +147,10 @@ extern void fmd_hdl_free(fmd_hdl_t *, void *, size_t);
 extern char *fmd_hdl_strdup(fmd_hdl_t *, const char *, int);
 extern void fmd_hdl_strfree(fmd_hdl_t *, char *);
 
-extern void fmd_hdl_vdebug(fmd_hdl_t *, const char *, va_list);
-extern void fmd_hdl_debug(fmd_hdl_t *, const char *, ...);
+extern void fmd_hdl_vdebug(fmd_hdl_t *, const char *, va_list)
+    __attribute__((format(printf, 2, 0)));
+extern void fmd_hdl_debug(fmd_hdl_t *, const char *, ...)
+    __attribute__((format(printf, 2, 3)));
```

Comment:
Some related cleanup was done in 5dbf6c5a66d3de65f1b0436c8e83097dd49ff059.  However, in that case we opted to add explicit `u_longlong_t` / `longlong_t` casts rather than modify the format specifier.  It'd be nice to stay consistent with that approach unless there's some significant advantage to using `PRIu64` / `PRId64`.  We have a few user space tools like `zdb`, `ztest`, and some test cases but they're currently the exception.

Reply:
Alright. I will revise this.